horace_core/sqw/@sqw/split.m

```diff
@@ -1,24 +1,77 @@
+% single spe or nxspe data set
 %
 %   >> wout = split(w)
 %
 % Input:
 % ------
 %   w        --  Input sqw object
+% Optional:
+%  folder_for_parts
+%            --  The string contains full path to the folder where to place
+%                sqw files representing parts of the sqw file to split.
+%                If absent the files will be placed in working directory.
+%
+%                If operation perfomed in memory only, this path is ignored.
+%                If this folder is provided, the resulting files are
+%                assumed permanent files, so are not getting deleted when
+%                their correspondent sqw objects are getting deleted from
+%                memory.
 % Optional keys:
 %  '-files'  -- if provided, return list of sqw files instead of sqw
 %               objects. When split objects do not fit memory, they are all
 %               stored in files and split returns list of the files.
-%               When the option is provided,
+%               When '-files' option is provided the split returns files
+%               instead of objects in any situation.
+%  '-filebacked'
+%           -- if provided, request split object to be filebacked
+%              regardless of the possibility to place them in memory.
+%              If split object' images do not fit memory, this option
+%              is ignored and the code behaves as if option '-files' is
+%              specified.
 %
 % Output:
 % -------
 %   wout    Array of sqw objects, each one made from a single spe data file
-%           If w is filebacked object, wout are filebacked too.
 %
-%           If their images do not fit to memory, wout would be the list of
-%           filenames, containing these files.
+%           If all images from split files do not fit memory, wout would be
+%           the list of filenames, referring to sqw files with contents of
+%           the split objects.
+% NOTE:
+% if results is filebacked or list of files, the resulting files or files-bases
+% for filebacked objects are placed in the 'folder_for_parts' directory or in
+% working directory if 'folder_for_path' is not provided.
+% The names of partial files are build from the name of the original sqw
+% file with added suffix containing corresponding run_id.
+%
+% For example: If you have initial sqw file Fe400mEv.sqw containing runs
+% 32400,32401 and 32402 and split it in filebacked mode, the folder
+% provided to keep result would contain files Fe400mEv_runID0032400.sqw,
+% Fe400mEv_runID0032401.sqw and Fe400mEv_runID0032402.sqw
+
+if numel(w)>1
+    error('HORACE:split:not_implemented', ...
+        'split currently works with only one sqw object');
+end
+[ok,mess,return_files,split_filebacked,argi] = parse_char_options(varargin, ...
+    {'-files','-filebacked'});
+if ~ok
+    error('HORACE:split:invalid_arguments',mess);
+end
+if ~isempty(argi)
+    folder_for_parts = argi{1};
+    if ~isfolder(folder_for_parts)
```

Comment:
What if it is a folder already? Is that bad?

Reply:
If folder is there, fine. It will use it to place result